### PR TITLE
Limit form variable substitutions to canonical-origin actions

### LINKED ISSUE #7112: Implement Var Sub in amp-form origin limitation
Similar to var-subs in anchors, amp-form should limit var-subs to the publishers origin action.

### PR DESCRIPTION
**Pull request by mkhatib**
Closes #7112

### CODE REVIEW

File: extensions/amp-form/0.1/amp-form.js [hunk 1]
```diff
@@ -255,6 +272,30 @@ export class AmpForm {
...
+   * @private
+   */
+  isSubmittingToCanonical_() {
+    if (this.isCanonicalAction_ !== undefined) {
+      return this.isCanonicalAction_;
+    }
+
+    const docInfo = documentInfoForDoc(this.form_);
+    const canonicalOrigin = parseUrl(docInfo.canonicalUrl).origin;
+    if (this.xhrAction_) {
```
Comment (jridgewell) [inline extensions/amp-form/0.1/amp-form.js]: Nit: Small duplication.

```js
const url = this.xhrAction_ || this.form.getAttribute('action');
this.isCanonicalAction_ = parseUrl(url).origin === canonicalOrigin;
```

File: extensions/amp-form/0.1/amp-form.js [hunk 2]
```diff
@@ -255,6 +272,30 @@ export class AmpForm {
...
+   * or not.
+   * @private
+   */
+  isSubmittingToCanonical_() {
+    if (this.isCanonicalAction_ !== undefined) {
+      return this.isCanonicalAction_;
+    }
+
+    const docInfo = documentInfoForDoc(this.form_);
+    const canonicalOrigin = parseUrl(docInfo.canonicalUrl).origin;
```
Comment (jridgewell) [inline extensions/amp-form/0.1/amp-form.js]: [`getSourceOrigin`](https://github.com/ampproject/amphtml/blob/master/src/url.js#L367-L375)?

Comment (mkhatib) [inline extensions/amp-form/0.1/amp-form.js]: Using `getSourceOrigin` might give false positives here - although that case is validated but I'd rather have a redundant use-case here. In general we don't allow `cdn.ampproject.org` urls to be a form action, so in this case we'd get the source origin although the URL would submit to `cdn.ampproject.org` origin.

File: extensions/amp-form/0.1/amp-form.js (hunk 1 above)
Comment (mkhatib) [inline extensions/amp-form/0.1/amp-form.js]: Done

**Pull request merged**